### DISCUSSION TITLE
tmp_path symlink oddity

### DISCUSSION THREAD

**Discussion by nicoddemus** [pytest-dev/pytest#10910] [General]
Hi folks,

While investigating #10896, I noticed something which I find odd.

To reiterate how things work: when we use a `tmp_path` fixture for the first time, we create the `basetemp` directory using `make_numbered_dir`, creating a new base temporary directory exclusive for that session:

```
E:\.tmp\pytest-of-bruno\pytest-905
        ^^^^^^^^^^^^^^^ ^^^^^^^^^^ 
        root dir        basetemp
```

That `root dir` is created once for each user, and `basetemp` is created for each pytest session.

In addition to that `pytest-905` directory, it will also create a symlink to it named `pytest-current` in the same directory (which I'm not sure what the use case is).

```
E:\.tmp\pytest-of-bruno\pytest-905
E:\.tmp\pytest-of-bruno\pytest-current  <- symlink to pytest-905
```

So when a test with `tmp_path` executes, the fixture creates a new temporary directory for that test inside `basetemp`, again using `make_numbered_dir`, which also includes that symlink:

```
E:\.tmp\pytest-of-bruno\pytest-908\test1_7_0
E:\.tmp\pytest-of-bruno\pytest-908\test1_7_current  <- symlink to test1_7_0
```

I kinda understand the reason for that first symlink to `basetemp` (`pytest-current`), but can anyone explain the reason why we have a symlink for *each* directory created by `tmp_path` (`test1_7_current`, above)?

**Comment by RonnyPfannschmidt** [comment]
It's a make numbered dir convenience going back to debugging pypy tests in pdb

We can probably drop it over printing the paths these days

**Reply by nicoddemus** [reply]
Thanks @RonnyPfannschmidt 

> We can probably drop it over printing the paths these days

Drop the top-level symlink (`pytest-905` in the example above), the symlinks to the test directories (`test_1_7_0`), or both?

**Reply by RonnyPfannschmidt** [reply]
Both
